flink-runtime/src/main/java/org/apache/flink/runtime/io/network/partition/consumer/SingleInputGate.java

```diff
@@ -219,6 +211,17 @@ public SingleInputGate(
 		this.isCreditBased = isCreditBased;
 	}
 
+	@Override
+	public void setup() throws IOException {
```

Comment:
In the setup, we should also `checkState(this.bufferPool == null)` firstly, then it could void bring potential trouble of releasing the related buffers if setup is called twice.